ita-social-projects/GreenCityUBS
# [UBS] Optimize creation ubs profile for user #5258

# GreenCityUBS PR
_&lt;Optimize creation ubs profile for user #5258&gt;_

## Summary Of Changes :fire:
_&lt;The creation of the UBS profile has been transferred from GreenCityUbs to GreenCityUser at the time of mail verification or sign up via Google.&gt;_

## Issue Link :clipboard:
_&lt;[Link to the issue](https://github.com/ita-social-projects/GreenCity/issues/5258)&gt;_

## Related PR in GreenCityUser :clipboard:
_&lt;https://github.com/ita-social-projects/GreenCityUser/pull/216&gt;_

## Added
* _&lt;Endpoint "/user/create" in UserProfileController&gt;_
* _&lt;UserProfileCreateDto&gt;_
* _&lt;Method createUserProfile in UBSClientService for creating ubs profile&gt;_

## Changed
* _&lt;Existing tests&gt;_

## Deleted
* _&lt;Method createUserByUuidIfUserDoesNotExist in UBSClientService &gt;_
* _&lt;Method calls createUserByUuidIfUserDoesNotExist &gt;_

## How to test :clipboard:
1st:
_&lt;Get GoogleId token&gt;_
_&lt;Sign up via swagger using GoogleId token&gt;_
_&lt;Sign up via swagger using GoogleId token&gt;_
_&lt;Check the successful registration, as well as the creation of a Ubs profile&gt;_
2nd:
_&lt;Sign up in GreenCity without Google using swagger&gt;_
_&lt;Verify email using swagger&gt;_
_&lt;Check the successful registration, as well as the creation of a Ubs profile&gt;_

# PR Checklist Forms

_(to be filled out by PR submitter)_
